stake-pool/cli/src/main.rs

```diff
@@ -374,120 +340,304 @@ fn command_create_pool(
+                        &spl_token::id(),
+                        &mint_keypair.pubkey(),
+                        withdraw_authority,
+                        None,
+                        default_decimals,
+                    )?,
+                ];
+                let signers = vec![config.fee_payer.as_ref()];
+                let transaction = checked_transaction_with_signers(config, &instructions, &signers)?;
+                println!("Initializing existing mint account {}", mint_keypair.pubkey());
+                send_transaction(config, transaction)?;
+                return Ok(());
+            }
+        }
+    }
+
+    let instructions = vec![
         system_instruction::create_account(
             &config.fee_payer.pubkey(),
             &mint_keypair.pubkey(),
             mint_account_balance,
             spl_token::state::Mint::LEN as u64,
             &spl_token::id(),
         ),
-        // Initialize pool token mint account
         spl_token::instruction::initialize_mint(
             &spl_token::id(),
             &mint_keypair.pubkey(),
-            &withdraw_authority,
+            withdraw_authority,
             None,
             default_decimals,
         )?,
     ];
 
-    let pool_fee_account = add_associated_token_account(
+    let signers = vec![config.fee_payer.as_ref(), mint_keypair];
+    let transaction = checked_transaction_with_signers(config, &instructions, &signers)?;
+
+    println!("Creating and initializing mint account {}", mint_keypair.pubkey());
+    send_transaction(config, transaction)?;
+    Ok(())
+}
+
+fn setup_pool_fee_account(
+    config: &Config,
+    mint_pubkey: &Pubkey,
+    total_rent_free_balances: &mut u64,
+) -> CommandResult {
+    let pool_fee_account = get_associated_token_address(&config.manager.pubkey(), mint_pubkey);
+    let pool_fee_account_info = config.rpc_client.get_account(&pool_fee_account);
+    if let Ok(account) = pool_fee_account_info {
+        if account.owner == spl_token::id() {
+            println!("Pool fee account {} already exists", pool_fee_account);
+            return Ok(());
+        }
+    }
+    // Create pool fee account
+    let mut instructions = vec![];
+    add_associated_token_account(
         config,
-        &mint_keypair.pubkey(),
+        mint_pubkey,
         &config.manager.pubkey(),
-        &mut setup_instructions,
-        &mut total_rent_free_balances,
+        &mut instructions,
+        total_rent_free_balances
     );
+
     println!("Creating pool fee collection account {}", pool_fee_account);
 
-    let initialize_instructions = &[
-        // Validator stake account list storage
-        system_instruction::create_account(
+    let signers = vec![config.fee_payer.as_ref()];
+    let transaction = checked_transaction_with_signers(config, &instructions, &signers)?;
+
+    send_transaction(config, transaction)?;
+    Ok(())
+}
+
+fn setup_and_initialize_validator_list_with_stake_pool(
```

Comment:
This needs a clippy allow according to CI
```suggestion

#[allow(clippy::too_many_arguments)]
fn setup_and_initialize_validator_list_with_stake_pool(
```